solr/core/src/java/org/apache/solr/core/SolrConfig.java

```diff
@@ -413,7 +413,7 @@ private void readXml(
       Map<String, IndexSchemaFactory.VersionedConfig> configCache,
       ResourceProvider rp)
       throws IOException {
-    XmlConfigFile xml = new XmlConfigFile(loader, rp, name, null, "/config/", null);
+    XmlConfigFile xml = new XmlConfigFile(loader, name, "/config/", null);
```

Comment:
What's going on with ResourceProvider `rp`?

Reply:
We can keep it here since we continue to consume its attributes

Reply:
I confirm this by code inspection.  From what I see, ResourceProvider should perhaps be nothing more than a POJO; it doesn't even _need_ methods.

CC @noblepaul -- you added a constructor to XmlConfigFile last October to take a `Function<String, InputStream> fileSupplier` that seems could simply be the InputStream directly.